test/e2e/e2e_test.go

```diff
@@ -66,9 +87,40 @@ var _ = Describe("Manager", Ordered, func() {
 		Expect(err).NotTo(HaveOccurred(), "Failed to install CRDs")
 
 		By("deploying the controller-manager")
+		projectImage := os.Getenv("IMG")
+		Expect(projectImage).ToNot(Equal(""), "IMG environment variable must be declared")
 		cmd = exec.Command("make", "deploy", fmt.Sprintf("IMG=%s", projectImage))
 		_, err = utils.Run(cmd)
 		Expect(err).NotTo(HaveOccurred(), "Failed to deploy the controller-manager")
+
+		By("Creating a k8s client")
+		k8sClient = getK8sClientOrDie(ctx)
+
+		By(fmt.Sprintf("Creating a namespace: %s", nsName), func() {
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nsName,
+				},
+			}
+			err := k8sClient.Create(ctx, ns)
+			if !(err == nil || kerrors.IsAlreadyExists(err)) {
+				Expect(err).ToNot(HaveOccurred())
+			}
```

Comment:
If you like it, we could rely on `Satisfy` here

```suggestion
			Expect(k8sClient.Create(ctx, ns)).To(Satisfy(func(err error) bool {
				return err == nil || kerrors.IsAlreadyExists(err)
			}
```